honeycomb/honeycomb.go

```diff
@@ -268,12 +268,13 @@ func honeycombSpan(s *export.SpanData) *Span {
 
 	hcSpan := &Span{
 		TraceID:         getHoneycombTraceID(sc.TraceIDString()),
-		ID:              fmt.Sprintf("%d", sc.SpanID),
+		ID:              sc.SpanIDString(),
 		Name:            s.Name,
 		HasRemoteParent: s.HasRemoteParent,
 	}
-	if s.ParentSpanID != sc.SpanID && s.ParentSpanID != 0 {
-		hcSpan.ParentID = fmt.Sprintf("%d", s.ParentSpanID)
+	parentID := hex.EncodeToString(s.ParentSpanID[:])
+	if s.ParentSpanID != sc.SpanID && parentID != "0000000000000000" {
```

Comment:
This check makes me sad. Is there a better way???

Reply:
check if s.ParentSpanID is equal to the empty default `[8]byte`? remember all Go values are 0-initialized...